cylc/flow/scripts/broadcast.py

```diff
@@ -196,13 +196,36 @@ def get_rdict(left, right=None, for_cancel_broadcast=False):
     upg(
         {'runtime': {'__MANY__': rdict}},
         'test',
-        for_cancel_broadcast=for_cancel_broadcast,
+        broadcast='cancel' if for_cancel_broadcast else True,
     )
+    rdict = strip_empty_sections(rdict)
     # Perform validation, but don't coerce the original (deepcopy).
     cylc_config_validate(deepcopy(rdict), SPEC['runtime']['__MANY__'])
     return rdict
 
 
+def strip_empty_sections(data):
+    """Recursively strip data structure of empty dicts.
+
+    This is needed post-upgrade as the upgrader can leave empty sections
```

Comment:
Can we not make the upgrade not do that? Or is it a feature that the upgrade actually uses?